qiita_db/portal.py

```diff
@@ -5,13 +5,66 @@
 #
 # The full license is in the file LICENSE, distributed with this software.
 # -----------------------------------------------------------------------------
+import warnings
 
 from .sql_connection import SQLConnectionHandler
 from .util import convert_to_id
+from .exceptions import QiitaDBError, QiitaDBDuplicateError
 from qiita_core.qiita_settings import qiita_config
+from qiita_core.exceptions import IncompetentQiitaDeveloperError
 
 # Only make functions available for main portal
 if qiita_config.portal == "QIITA":
+    def create_portal(portal, desc):
```

Comment:
This totally looks now like it should be an object... It has create, delete, properties...
At the end you implemented everything that an object requires. I don't understand why you didn't want at the beginning to implement the object... This type of (bad) decisions is what makes Qiita use up a lot of developer time. I can see in the future doing a refactoring of this code to make it an object as this is what it fits, some time that we could avoid if this was an object since the very beginning...